src/engine/execute.js

```diff
@@ -207,7 +219,19 @@ const execute = function (sequencer, thread) {
     let primitiveReportedValue = null;
     blockUtility.sequencer = sequencer;
     blockUtility.thread = thread;
+    if (runtime.profiler !== null) {
+        if (blockFunctionProfilerId === -1) {
+            blockFunctionProfilerId = runtime.profiler.idByName(blockFunctionProfilerFrame);
+        }
+        // runtime.profiler.start(blockFunctionProfilerId, opcode);
+        runtime.profiler.records.push(
+            runtime.profiler.START, blockFunctionProfilerId, opcode, performance.now());
```

Comment:
This is the code of `Profiler.start` manually inlined. For small functions in jS, the virtual machines can end up doing more to make the call than perform the logic in the call. Its one of those techniques you want to balance for sections of JavaScript that are critical to performance.

Its true for start and stop, but only need it here and the calls around the call to execute in Sequencer to have the recorded information better reflect normal performance instead of having added bias over time from `Profiler.start`.

This is one of the things I'd like to do through a babel plugin like I briefly mentioned yesterday, to separate the performance concern into a different part of the scratch-vm code through code generation.